internal/kafka/internal/services/kafka.go

```diff
@@ -1428,15 +1447,11 @@ func (k *kafkaService) ManagedKafkasRoutesTLSCertificate(kafkaRequest *dbapi.Kaf
 		return err
 	}
 
-	values := map[string]interface{}{
-		"kafkas_routes_base_domain_name":        kafkaRequest.KafkasRoutesBaseDomainName,
-		"kafkas_routes_base_domain_tls_crt_ref": certManagementOutput.TLSCertRef,
-		"kafkas_routes_base_domain_tls_key_ref": certManagementOutput.TLSKeyRef,
-	}
+	kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef = certManagementOutput.TLSKeyRef
+	kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef = certManagementOutput.TLSCertRef
 
-	dbConn := k.connectionFactory.New().Model(kafkaRequest)
-	if err := dbConn.Updates(values).Error; err != nil {
-		return errors.NewWithCause(errors.ErrorGeneral, err, "failed to update kafka %q", kafkaRequest.ID)
+	if svcErr := k.Update(kafkaRequest); svcErr != nil {
```

Comment:
Is there a possibility that the kafkaRequest contains non-zero valued fields that are different than what's in the DB aside from the ones being set in this method?

Reply:
At the moment no, but it could be possible in the future. I wanted the convinience of using the `kafkaRequest` but I guess I'll need to change it back and make it more specific as it was initially.

Reply:
changed back to what it was before

Reply:
Another alternative could be creating an empty kafkarequest, copying the desired values there and then use that one to perform the update. Although I haven't tried it

Reply:
Yes. That would also work. I've changed it back so I'll keep it as it.